streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -1468,9 +1456,6 @@ public void shouldCloseActiveTasksAndIgnoreExceptionsOnUncleanShutdown() {
 
         resetToStrict(changeLogReader);
         expect(changeLogReader.completedChangelogs()).andReturn(emptySet());
-        // make sure we also remove the changelog partitions from the changelog reader
-        changeLogReader.remove(eq(singletonList(changelog)));
-        expectLastCall();
```

Comment:
I feel like I missed something here. We don't expect the changelog to get unregistered during close anymore?

Reply:
The "unregistration" is now handled by the ProcessorStateManager, which is mocked in this test. There are new tests in ProcessorStateManagerTest that verify we unregister the partitions during close/recycle

Reply:
Thanks.